### DISCUSSION TITLE
connect lambda to API Gateway

### DISCUSSION THREAD

**Discussion by corentinway** [awslabs/aws-sdk-rust#754] [Q&A] [answered]
Is it possible to do this by rust code : 
1. create a lambda (yes with cargo-lambda)
2. create an API gateway
3. connect the lambda to the API gateway ?

Please advise

Regards

**Comment by jmklix** [comment, marked as answer]
1. Yes with `create_function` [here](https://docs.rs/aws-sdk-lambda/latest/aws_sdk_lambda/client/struct.Client.html#method.create_function)
As long as the lambda code is [supported](https://aws.amazon.com/lambda/faqs/):
> AWS Lambda natively supports Java, Go, PowerShell, Node.js, C#, Python, and Ruby code

2. Yes. [API gateway](https://docs.rs/aws-sdk-apigateway/latest/aws_sdk_apigateway/client/struct.Client.html)
3. I think you are looking for `create_authorizer` [here](https://docs.rs/aws-sdk-apigateway/latest/aws_sdk_apigateway/client/struct.Client.html#method.create_authorizer)

**Reply by corentinway** [reply]
Thank you for your proposal.

My code here  https://gist.github.com/corentinway/f26fd21a113287de74cb97b92bfac38e 

is giving me the error : 

```
thread 'main' panicked at 'There is an issue: ServiceError(ServiceError { source: PutIntegrationError { kind: NotFoundException(NotFoundException { message: Some("Invalid Method identifier specified") }), meta: Error { code: Some("NotFoundException"), message: Some("Invalid Method identifier specified"), request_id: Some("d94e6801-f5a3-4c8f-9540-961a3fa3ad8e"), extras: {} } }, raw: Response { inner: Response { status: 404, version: HTTP/2.0, headers: {"date": "Thu, 16 Mar 2023 13:56:57 GMT", "content-type": "application/json", "content-length": "50", "x-amzn-requestid": "d94e6801-f5a3-4c8f-9540-961a3fa3ad8e", "x-amzn-errortype": "NotFoundException", "x-amz-apigw-id": "B4H2dKLOiGYEdQg="}, body: SdkBody { inner: Once(Some(b"{\"message\":\"Invalid Method identifier specified\"}\n")), retryable: true } }, properties: SharedPropertyBag(Mutex { data: PropertyBag, poisoned: false, .. }) } })', src/main.rs:12:32
```

==> `Invalid Method identifier specified`  


Please advise
